modules/app/src/main/java/org/locationtech/jtstest/function/SpatialIndexFunctions.java

```diff
@@ -117,6 +117,14 @@ public static Geometry strTreeNN(Geometry geoms, Geometry geom)
     return (Geometry) result;
   }
 
+  public static Geometry[] strTreeNNk(Geometry geoms, Geometry geom)
+  {
+    STRtree index = buildSTRtree(geoms);
+    int k = 10;
+    Object[] result = index.nearestNeighbour(geom.getEnvelopeInternal(), geom, new GeometryItemDistance(), k);
+    return (Geometry[]) result;
```

Comment:
@dr-jts I converted them to GeometryCollection manually because the TestBuilder doesn't take Geometry[].